pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -461,13 +466,19 @@ def _set_accelerator_if_ipu_strategy_is_passed(self) -> None:
         if isinstance(self._strategy_flag, IPUStrategy):
             self._accelerator_flag = "ipu"
 
+    def _set_accelerator_if_hpu_strategy_is_passed(self) -> None:
+        if isinstance(self._strategy_flag, SingleHPUStrategy):
+            self._accelerator_flag = "hpu"
+
```

Comment:
1. `_set_accelerator_if_hpu_strategy_is_passed` and `_set_accelerator_if_ipu_strategy_is_passed` could be merge to `_set_accelerator_base_on_strategy` with comments "The strategy user passed in only work with x_accelerator, so set accelerator to x"
2. Only SingleHPUStrategy has to run on hpu? How about HPUParallelStrategy?
3. Overall I don't recommend this logic. We kept `_set_accelerator_if_ipu_strategy_is_passed` for IPU as an exception (for backward compatibility), other accelerator type doesn't have "choose accelerator base on strategy" logic. We should either apply this to all applicable strategy/accelerator type or don't do it at all. I think it's on user to pass the correct `accelerator`. If doesn't compatible, we should throw exceptions instead of silently change it, as we don't know what does user wants. For example, user pass `accelerator=cpu, strategy=`ipu`, we by default silently used ipu, but maybe user wants run a test on cpu, just passed wrong strategy. cc @kaushikb11 @awaelchli

Reply:
Fair point! @four4fish I have removed it.